src/DotNetCore.CAP/Messages/Message.cs

```diff
@@ -9,16 +9,21 @@ namespace DotNetCore.CAP.Messages
 {
     public class Message
     {
+        /// <summary>
+        /// System.Text.Json requires that class explicitly has a parameterless constructor
+        /// </summary>
+        public Message() {}
+
         public Message(IDictionary<string, string> headers, [CanBeNull] object value)
         {
             Headers = headers ?? throw new ArgumentNullException(nameof(headers));
             Value = value;
         }
 
-        public IDictionary<string, string> Headers { get; }
+        public IDictionary<string, string> Headers { get; set; }
```

Comment:
I'm not sure a public set is wanted. The same for Value

Reply:
The system.text.json serializer requires a parameterless constructor and seems to skip properties without a set;

That was the problem we faced when we changed this to set;

Any better ideas?

Reply:
I'd forgotten that. But you could create a custom serializer for the class (or switch to .net 5.0, it's possible there).